Makefile

```diff
@@ -2,6 +2,7 @@ BUILD_DIR=$(shell pwd)/build
 HANDLE_USER_DATA=$(shell base64 -w 0 scripts/handle-user-data)
 CERT_GEN=$(shell base64 -w 0 scripts/cert-gen.sh)
 VERSION=1.0.0-alpha.1
+TODAY=$(shell date +"%d-%m-%Y-%H-%M-%S")
```

Comment:
Whatever really. sha-date-build-id works. Personally, I drop the dashes in the date. Just as single string. I think this makes it easier on the eye and also separates the parts better.

Regarding the sha (maybe 6 to 8 characters), as @LalatenduMohanty I think it is a nice to have, but not a must. Should be easy to get though. The injected build id is imo a must have and we will use/need it sooner or later. 

What to use for the case that no build id is set in the environment, I have no strong opinion. As said, _local_ might be a solution or one just leaves it off.